tenders/hvt/hvt.h

```diff
@@ -107,6 +107,13 @@ void hvt_vcpu_init(struct hvt *hvt, hvt_gpa_t gpa_ep);
 void hvt_boot_info_init(struct hvt *hvt, hvt_gpa_t gpa_kend, int cmdline_argc,
         char **cmdline_argv, struct mft *mft, size_t mft_size);
 
+/*
+ * Apply page protections to guest memory. See guest_mprotect_fn_t in elf.h for
+ * a full description.
+ */
+int hvt_guest_mprotect(void *t, uint64_t addr_start, uint64_t addr_end,
```

Comment:
Would it be tricky to just do `struct hvt *t` directly?
```suggestion
int hvt_guest_mprotect(struct hvt *t, uint64_t addr_start, uint64_t addr_end,
```

Reply:
It seems like `addr_start` and `addr_end` could additionally be const? (To decrease risk of confusing `vaddr_start` and `addr_start` when the former is assigned)

Reply:
> It seems like addr_start and addr_end could additionally be const? (To decrease risk of confusing vaddr_start and addr_start when the former is assigned)

That doesn't work well in C for primitive types, since if you have e.g.

    int foo(const int bar);

then whatever `int` the _caller_ passed to `foo()` must also be `const` on the caller's side. So it's not very practical. Not worth it here.